website/profile/views.py

```diff
@@ -786,31 +786,36 @@ def request_deactivation(auth):
     return {'message': 'Sent account deactivation request'}
 
 
-def redirect_to_twitter(uid):
-    """Redirect GET requests for /@uid/ to respective the Twitter account if a User
-    exists and has a Twitter handle associated with the account.
+def redirect_to_twitter(twitter_handle):
+    """Redirect GET requests for /@TwitterHandle/ to respective the OSF user
+    account if it associated with an active account
 
     :param uid: uid for requested User
     :return: Redirect to User's Twitter account page
     """
-    user = User.load(uid)
-
-    if user:
-        twitter_handle = user.social.get('twitter', None)
-    else:
+    try:
+        user = User.find_one(Q('social.twitter', 'iexact', twitter_handle))
+    except NoResultsFound:
         raise HTTPError(http.NOT_FOUND, data={
             'message_short': 'User Not Found',
-            'message_long': 'There is no active user associated with user id: {0}.'.format(uid)
+            'message_long': 'There is no active user associated with the Twitter handle: {0}.'.format(twitter_handle)
         })
-
-    if twitter_handle:
-        # TODO(hrybacki): Verify Twitter handle is for a real account.
-        # Requires authenticated access to Twitter API
-        redirect_url = "https://twitter.com/{0}".format(twitter_handle)
-    else:
-        raise HTTPError(http.BAD_REQUEST, data={
-            'message_short': 'Invalid Request',
-            'message_long': '{0} does not have a Twitter handle associated with their account.'.format(user.fullname)
+    except MultipleResultsFound:
+        users = User.find(Q('social.twitter', 'iexact', twitter_handle))
+        message_long = 'There are multiple OSF accounts associated with the ' \
+                       'Twitter handle: <strong>{0}</strong>. <br /> Please ' \
+                       'select from the accounts below. <br /><ul>'.format(twitter_handle)
+        for user in users:
+            message_long += '<li><a href="{0}">{1}</a></li>'.format(web_url_for(
+                'profile_view_id',
+                uid=user._id),
+                user.fullname
+            )
+        message_long += '</ul>'
+        raise HTTPError(http.MULTIPLE_CHOICES, data={
```

Comment:
Ah, after reading the rfc, it seems to be that automatic redirection is not required. So this might be ok.

> Unless it was a HEAD request, the response SHOULD include an entity containing a list of resource characteristics and location(s) from which the user or user agent can choose the one most appropriate. The entity format is specified by the media type given in the Content- Type header field. Depending upon the format and the capabilities of the user agent, selection of the most appropriate choice MAY be performed automatically. 

So this is fine as is, so long as browsers don't produce any unwanted behavior, such as opening a prompt.